src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,356 @@
+  /** The value to which the slider thumb should be rounded */
+  roundToValue?: number
+
+  /** custom element that can be used to replace the default thumb of the slider */
+  thumbComponent?: React.ReactElement
+
+  /** Callback function to handle the change in slider value */
+  onValueChange?: (value: number) => void
+}
+
+export interface SliderCommonProps {
+  /** The maximum value of the slider */
+  maximumValue?: number
+
+  /** The minimum value of the slider */
+  minimumValue?: number
+
+  /** The step value for the slider */
+  step?: number
+
+  /** The alwaysShowValue indicates whether the value of the slider should always be displayed */
+  alwaysShowValue?: boolean
+
+  /** Whether to show the point on the slider's track */
+  showTrackPoint?: boolean
+
+  /** Determines whether the thumb can be moved by directly touching the thumb or only by dragging the slider track */
+  tapToSeek?: boolean
+
+  /** The touchable area is used to increase the size of the thumb and make it easier to interact with */
+  hitSlopPoint?: Insets | number
+
+  /** Style for the slider component. */
+  style?: StyleProp<ViewStyle>
+
+  /** Style of the slider's track */
+  trackStyle?: StyleProp<ViewStyle>
+
+  /** Style for the track's filled portion (in case of a range slider) */
+  trackedStyle?: StyleProp<ViewStyle>
+
+  /** Style of the point on the slider's track */
+  trackPointStyle?: StyleProp<ViewStyle>
+
+  /** The bgColorLabelView sets the background color of the view that displays the value of the slider */
+  bgColorLabelView?: string
+
+  /** The labelStyle sets the style of the text that displays the value of the slider */
+  labelStyle?: StyleProp<TextStyle>
+
+  /** Style of the slider's thumb */
+  thumbStyle?: StyleProp<ViewStyle>
+
+  /** Size of the slider's thumb */
+  thumbSize?: Size
+}
+
+type SliderPropsWithOptionalWidth = {sliderWidth?: number} & (
+  | {showTrackPoint: true; sliderWidth: number}
+  | {showTrackPoint?: false; sliderWidth?: number}
+)
+
+type SliderProps = SliderCommonProps & ISlider & SliderPropsWithOptionalWidth
+
+type SliderComponentProps = React.FC<SliderProps> & {
+  Range: React.FC<SliderRangeProps>
+  FixedRange: React.FC<SliderFixedRangeProps>
+  Fixed: React.FC<SliderProps>
+}
+
+const Slider: SliderComponentProps = ({
+  minimumValue = DEFAULT_MINIMUM_VALUE,
+  maximumValue = DEFAULT_MAXIMUM_VALUE,
+  step,
+  roundToValue,
+  style,
+  trackStyle,
+  trackedStyle,
+  thumbStyle,
+  bgColorLabelView,
+  alwaysShowValue,
+  labelStyle,
+  thumbComponent,
+  showTrackPoint,
+  tapToSeek,
+  hitSlopPoint = hitSlop,
+  sliderWidth,
+  thumbSize = {width: metrics.medium, height: metrics.medium},
+  trackPointStyle,
+  onValueChange = () => null,
+}: SliderProps) => {
+  const sliderInfo = useSharedValue<SliderInfo>({
+    range: INIT_VALUE,
+    trackWidth: INIT_VALUE,
+  })
+  const sliderValue = useSharedValue<number>(INIT_POINT)
+  const progress = useSharedValue<number>(INIT_VALUE)
```

Comment:
Why we don't create just 1 useSharedValue for all related to slider?
Also how about change to **SliderState**  instead for correct meaning , for example:
```suggestion
  const sliderState = useSharedValue<SliderState>({
    range: INIT_VALUE,
    trackWidth: INIT_VALUE,
    currentValue: INIT_POINT,
    progress: INIT_VALUE
  })
```

Reply:
sliderState and sliderValue + progress is using for different purpose. So i think it more clearly to keep it separate like above